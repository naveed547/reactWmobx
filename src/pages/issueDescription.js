import React, { Component } from 'react';
//import {bindActionCreators} from "redux";
//import {connect} from "react-redux";
//import * as actions from "../actions/repoAction";
import moment from 'moment';
import '../components/description.scss';
import { observer, inject } from 'mobx-react';

@inject("RepoStore","UserStore")

@observer
export class IssueDescription extends Component {
	constructor(props) {
		super(props);
		
		this.RepoStore = this.props.RepoStore;
		this.UserStore = this.props.UserStore;
	}
  	renderIssue = () => {
		let issue = this.RepoStore.repoData; 
		return (
			<div className="card">
				<div className="card-header row mx-0">
					<div className="col-sm-10 pl-0">#{issue.id} <b>{issue.title}</b></div>
					<div className={issue.state === 'open' ? 'col-sm-2 text-right text-danger pr-0':'col-sm-2 text-right text-success pr-0'}>
						<i className={issue.state === 'open' ? 'fa fa-exclamation-circle text-danger':'fa fa-check text-success'}></i>
						<span className="pl-1 text-capitalize">{issue.state}</span>
					</div>	
				</div>
				<div className="card-body">
					<p className="card-text">Issue Number: <b>{issue.number}</b></p>
					<p className="card-text">{issue.body}</p>
				</div>
				<div className="card-body row">
					<p className="card-text col-sm-6">
						<span>Created By: </span>
						<span><b>{issue.user.login}</b></span>
					</p>
					<p className="card-text col-sm-6 text-right">
						<small className="text-muted">Last updated <b>{moment(issue.state === 'open' ? issue.updated_at : issue.closed_at).fromNow()}</b>
						</small>
					</p>
				</div>
			</div>
		);
  }
  componentDidMount() {
	this.RepoStore.loadIssue(this.props.match.params.issueId,this.props.match.params.user,this.props.match.params.repo);
  }
  render() {
  	if (this.RepoStore 
  		&& this.RepoStore.repoData 
  		&& this.RepoStore.repoData.number) {
	    return (
	    	<div>
	      		<h2 className="my-5 text-center">Issue Description</h2>
	      		User Name: {this.RepoStore.userName} <br/>
	      		Repo Name: {this.RepoStore.currentRepo} <br/>
	      		{this.renderIssue()}
	      	</div>
	    );
	}else {
		return (
			<div className="my-5 text-center">Loading...</div>
		)
	}
  }
}


/* const mapStateToProps = state => {
    return {
        repoDetails: state.reposReducer
    }
};

const mapDispatchToProps = dispatch => {
    return bindActionCreators(actions, dispatch);
}


const IssueDescriptionContainer = connect(mapStateToProps, mapDispatchToProps)(IssueDescription);


export default IssueDescriptionContainer; */

