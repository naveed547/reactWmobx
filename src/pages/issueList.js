import React, { Component } from 'react';
//import {bindActionCreators} from "redux";
//import {connect} from "react-redux";
import '../components/list.scss';
//import {ISFMapper,TFMapper,LTMapper} from '../helper/stateActionMapper';
//import * as actions from "../actions/repoAction";
import InputSearchFilter from '../components/inputSearchFilter';
import TopFilter from '../components/topFilter';
import IssueTable from '../components/listTable';
import { observer, inject } from 'mobx-react';

@inject("RepoStore","UserStore")

@observer
export class IssueList extends Component {
	constructor(props) {
		super(props);
		
		this.RepoStore = this.props.RepoStore;
		this.UserStore = this.props.UserStore;
	}
	componentDidMount() {
		this.RepoStore.loadIssues(this.props.match.params.repo,this.props.match.params.user);
	}
	render() {
		if (this.RepoStore.repoData.total_count !== undefined) {
		    return (
				<React.Fragment>
					<h2 className="text-center my-5">Issues List <span className="my-5 text-center">{this.RepoStore.repoData.total_count ? `(${this.RepoStore.repoData.total_count})` : ''} </span></h2>
					User Name: <span className="user-name">{this.RepoStore.userName}</span> <br/>
	      			Repo Name: <span className="repo-name">{this.RepoStore.currentRepo}</span> <br/>
	      			{this.RepoStore.repoData.total_count ? (
				        <div className="card">
							<div className="card-header row mx-0">
								<div className="col-sm-4">
									<InputSearchFilter search={this.UserStore.searchIssues} />
								</div>
								<div className="col-sm-8">
									<TopFilter />
								</div>
							</div>
							<IssueTable />	
						</div>
				      ) : (
				        <div className="my-5 text-center">No data found for this user</div>
				      )}
				</React.Fragment>
		    );
		}
		else {
			return (
				<div className="my-5 text-center">Loading...</div>
			)
		}
	}

  
}