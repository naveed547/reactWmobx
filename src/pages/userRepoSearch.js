import React, { Component } from 'react';
import '../components/description.scss';
import { Link } from 'react-router-dom';
import { observer, inject } from 'mobx-react';

@inject("RepoStore")

@observer 
class UserRepoSearch extends Component {
  constructor(props) {
    super(props);
		this.userName = React.createRef();
		
		this.store = this.props.RepoStore;
  }
  selectRepo = event => {
  	event.preventDefault();
  	const user = this.userName.current.value;
    this.store.loadRepos(user);
  }
  componentDidMount() {
		this.store.loadRepos(this.store.userName);
  }
  renderForm = () => {
      return (
		<form name="repoSearch" className="form-inline justify-content-center" onSubmit={(e) => {e.preventDefault();this.store.loadRepos(this.userName.current.value)}}>
		  <div className="form-group mb-2 mr-2">
		    <label htmlFor="userName" className="sr-only">Password</label>
		    <input type="text" className="form-control" id="userName" onChange={() => {}} name="user" ref={this.userName} value={this.store.userName} placeholder="Search Git user name" />
		  </div>
		  <button type="submit" className="btn btn-primary mb-2">Search</button>
		</form>
      );
  }
  renderRepoList() {
  	if (!this.store.repoData.length) {
  		return (<div>Loading...</div>);
  	} else {
  		return this.store.repoData.map((repo,index) => {
  			return (
  				<li className="list-group-item" key={repo.id}>
						<Link to={`/issueList/${this.store.userName}/${repo.name}`}>
							{repo.name}
						</Link>
  				</li>
  			)
  		})
  	}
  }
  render() {
  	return (
			<div>
				<h2 className="my-5 text-center">Search User and Repo</h2>
				User Name: <span className="user-name">{this.store.userName}</span> <br/>
				<div className="card">
					<div className="card-header">
						 {this.renderForm()} 
					</div>
					<ul className="list-group list-group-flush repo-list">{this.renderRepoList()}</ul>
				</div>
			</div>
	    );
  } 
}
export default UserRepoSearch;

