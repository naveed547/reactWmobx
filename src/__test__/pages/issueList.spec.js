import React from "react";
import {mount} from "enzyme";
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import {IssueList} from "../../pages/IssueList";
import initialState from '../../reducers/initialState';
import repoMock from "../data/mock";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";

describe("IssueList", () => {
    let mapData = Object.assign({},repoMock.allIssues);
        mapData['open_count'] = repoMock.openIssues.total_count;
        mapData['closed_count'] = mapData.total_count - mapData.open_count;
    const mockStore = configureMockStore();
    const store = mockStore({reposReducer: {
        userName: 'twbs',
        currentRepo: 'bootstrap',
        repoData:   mapData
    },
    userReducer: {
        ...initialState.filter
    }}); 
    it('IssueList component renders correctly', () => {
        /* let mapData = Object.assign({},repoMock.allIssues);
        mapData['open_count'] = repoMock.openIssues.total_count;
        mapData['closed_count'] = mapData.total_count - mapData.open_count; */
        const props = {
            loadIssues: jest.fn(),
            repoDetails: {
                ...initialState.repoDetails,
                repoData:   mapData,
            },
            count: mapData.total_count,
            match:{
                params:{
                    repo: repoMock.repoName,
                    user: repoMock.userName
                }
            }
        };
        const rendered = renderer.create(
            <Provider store={store}>
            <MemoryRouter>
                <IssueList {...props} />
            </MemoryRouter>
            </Provider>
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
    it("displays the IssueList component when it has fetched users", () => {
        /* let mapData = Object.assign({},repoMock.allIssues);
        mapData['open_count'] = repoMock.openIssues.total_count;
        mapData['closed_count'] = mapData.total_count - mapData.open_count; */
        const props = {
            loadIssues: jest.fn(),
            repoDetails: {
                ...initialState.repoDetails,
                repoData:   mapData,
            },
            count: mapData.total_count,
            match:{
                params:{
                    repo: repoMock.repoName,
                    user: repoMock.userName
                }
            }
        };
        const container = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <IssueList {...props} />
                </MemoryRouter>
            </Provider>
        );
        expect(container.find('.user-name').text()).toEqual(props.repoDetails.userName);
        expect(container.find('.repo-name').text()).toEqual(props.repoDetails.currentRepo);
    });
  
    it("does not display the IssueList when ther are no users", () => {
       /*  let mapData = Object.assign({},repoMock.allIssues);
        mapData['open_count'] = repoMock.openIssues.total_count;
        mapData['closed_count'] = mapData.total_count - mapData.open_count; */
        const props = {
            loadIssues: jest.fn(),
            repoDetails: {
                userName: ' ',
                repoData:   [],
                repoName: ' '
            },
            count: 0,
            match:{
                params:{
                    repo: repoMock.repoName,
                    user: repoMock.userName
                }
            }
        };
        
        const container = mount(
            <Provider store={store}>
                <MemoryRouter>
                   <IssueList {...props} />   
                </MemoryRouter>
            </Provider>
        );
        expect(container.find('.user-name').text()).toEqual(' ');
        expect(container.find('.repo-name').text()).toEqual('');
    });
  });
  