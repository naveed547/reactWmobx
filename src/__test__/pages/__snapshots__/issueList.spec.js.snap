// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueList IssueList component renders correctly 1`] = `
Array [
  <h2
    className="text-center my-5"
>
    Issues List 
    <span
        className="my-5 text-center"
    >
        (17734)
         
    </span>
</h2>,
  "User Name: ",
  <span
    className="user-name"
>
    twbs
</span>,
  " ",
  <br />,
  "Repo Name: ",
  <span
    className="repo-name"
>
    bootstrap
</span>,
  " ",
  <br />,
  <div
    className="card"
>
    <div
        className="card-header row mx-0"
    >
        <div
            className="col-sm-4"
        >
            <form
                id="searchForm"
                onSubmit={[Function]}
            >
                <input
                    className="form-control"
                    name="search"
                    onChange={[Function]}
                    type="search"
                />
            </form>
        </div>
        <div
            className="col-sm-8"
        >
            <div
                className="topFilter row"
            >
                <div
                    className="col text-center"
                >
                    <button
                        className="btn btn-secondary"
                        onClick={[Function]}
                        type="button"
                    >
                        <i
                            className="fa fa-exclamation-circle text-danger"
                        />
                         
                        366
                         Open
                    </button>
                    <button
                        className="btn btn-outline-secondary ml-1"
                        onClick={[Function]}
                        type="button"
                    >
                        <i
                            className="fa fa-check text-success"
                        />
                         
                        17368
                         Closed
                    </button>
                </div>
                <div
                    className="col text-right"
                >
                    <select
                        className="form-control"
                        onChange={[Function]}
                        value="Oldest"
                    >
                        <option>
                            Newest
                        </option>
                        <option>
                            Oldest
                        </option>
                        <option>
                            Most Commented
                        </option>
                        <option>
                            Least Commented
                        </option>
                    </select>
                </div>
            </div>
        </div>
    </div>
    <ul
        className="list-group list-group-flush"
    >
        <li
            className="list-group-item"
        >
            <div>
                <i
                    className="fa fa-exclamation-circle text-danger"
                />
                <a
                    className="issue-title"
                    href="/issue/twbs/bootstrap/27291"
                    onClick={[Function]}
                >
                      
                    Horizontal list group feature suggestion
                </a>
            </div>
            <span
                className="issue-desc"
            >
                #
                27291
                 
                opened
                 
                13 days ago
                 by 
                sts-ryan-holton
                <i
                    className="fa fa-comments pl-1"
                />
                 
                0
            </span>
            <span
                className="pl-2"
            >
                Last updated at 
                13 days ago
            </span>
        </li>
        <li
            className="list-group-item"
        >
            <div>
                <i
                    className="fa fa-exclamation-circle text-danger"
                />
                <a
                    className="issue-title"
                    href="/issue/twbs/bootstrap/27296"
                    onClick={[Function]}
                >
                      
                    Create a scroll-able utility class
                </a>
            </div>
            <span
                className="issue-desc"
            >
                #
                27296
                 
                opened
                 
                12 days ago
                 by 
                sts-ryan-holton
                <i
                    className="fa fa-comments pl-1"
                />
                 
                0
            </span>
            <span
                className="pl-2"
            >
                Last updated at 
                12 days ago
            </span>
        </li>
        <li
            className="list-group-item"
        >
            <div>
                <i
                    className="fa fa-exclamation-circle text-danger"
                />
                <a
                    className="issue-title"
                    href="/issue/twbs/bootstrap/27317"
                    onClick={[Function]}
                >
                      
                    Popover not triggered on long press on a button in Firefox mobile
                </a>
            </div>
            <span
                className="issue-desc"
            >
                #
                27317
                 
                opened
                 
                9 days ago
                 by 
                curtisy1
                <i
                    className="fa fa-comments pl-1"
                />
                 
                9
            </span>
            <span
                className="pl-2"
            >
                Last updated at 
                9 days ago
            </span>
        </li>
        <li
            className="list-group-item"
        >
            <div>
                <i
                    className="fa fa-exclamation-circle text-danger"
                />
                <a
                    className="issue-title"
                    href="/issue/twbs/bootstrap/27312"
                    onClick={[Function]}
                >
                      
                    How to make visible border lines of table header with z-index? (v.4) 
                </a>
            </div>
            <span
                className="issue-desc"
            >
                #
                27312
                 
                opened
                 
                9 days ago
                 by 
                MikaelBox
                <i
                    className="fa fa-comments pl-1"
                />
                 
                0
            </span>
            <span
                className="pl-2"
            >
                Last updated at 
                9 days ago
            </span>
        </li>
        <li
            className="list-group-item"
        >
            <div>
                <i
                    className="fa fa-exclamation-circle text-danger"
                />
                <a
                    className="issue-title"
                    href="/issue/twbs/bootstrap/27320"
                    onClick={[Function]}
                >
                      
                    Adapt our npm scripts for Netlify previews
                </a>
            </div>
            <span
                className="issue-desc"
            >
                #
                27320
                 
                opened
                 
                9 days ago
                 by 
                XhmikosR
                <i
                    className="fa fa-comments pl-1"
                />
                 
                0
            </span>
            <span
                className="pl-2"
            >
                Last updated at 
                9 days ago
            </span>
        </li>
        <li
            className="list-group-item"
        >
            <div>
                <i
                    className="fa fa-exclamation-circle text-danger"
                />
                <a
                    className="issue-title"
                    href="/issue/twbs/bootstrap/27311"
                    onClick={[Function]}
                >
                      
                    Shouldn't nested tables get a background color of $table-bg instead of $body-bg?
                </a>
            </div>
            <span
                className="issue-desc"
            >
                #
                27311
                 
                opened
                 
                9 days ago
                 by 
                johnaau
                <i
                    className="fa fa-comments pl-1"
                />
                 
                3
            </span>
            <span
                className="pl-2"
            >
                Last updated at 
                9 days ago
            </span>
        </li>
        <li
            className="list-group-item"
        >
            <div>
                <i
                    className="fa fa-exclamation-circle text-danger"
                />
                <a
                    className="issue-title"
                    href="/issue/twbs/bootstrap/27326"
                    onClick={[Function]}
                >
                      
                    Cannot set property 'emulateTransitionEnd' of undefined
                </a>
            </div>
            <span
                className="issue-desc"
            >
                #
                27326
                 
                opened
                 
                9 days ago
                 by 
                miracle0403
                <i
                    className="fa fa-comments pl-1"
                />
                 
                1
            </span>
            <span
                className="pl-2"
            >
                Last updated at 
                9 days ago
            </span>
        </li>
        <li
            className="list-group-item"
        >
            <div>
                <i
                    className="fa fa-exclamation-circle text-danger"
                />
                <a
                    className="issue-title"
                    href="/issue/twbs/bootstrap/27327"
                    onClick={[Function]}
                >
                      
                    Form input does not inherit font-weight
                </a>
            </div>
            <span
                className="issue-desc"
            >
                #
                27327
                 
                opened
                 
                8 days ago
                 by 
                brnrd87
                <i
                    className="fa fa-comments pl-1"
                />
                 
                1
            </span>
            <span
                className="pl-2"
            >
                Last updated at 
                8 days ago
            </span>
        </li>
        <li
            className="list-group-item"
        >
            <div>
                <i
                    className="fa fa-exclamation-circle text-danger"
                />
                <a
                    className="issue-title"
                    href="/issue/twbs/bootstrap/27333"
                    onClick={[Function]}
                >
                      
                    Padding added when modal is available
                </a>
            </div>
            <span
                className="issue-desc"
            >
                #
                27333
                 
                opened
                 
                8 days ago
                 by 
                TechnicalSuwako
                <i
                    className="fa fa-comments pl-1"
                />
                 
                0
            </span>
            <span
                className="pl-2"
            >
                Last updated at 
                8 days ago
            </span>
        </li>
        <li
            className="list-group-item"
        >
            <div>
                <i
                    className="fa fa-exclamation-circle text-danger"
                />
                <a
                    className="issue-title"
                    href="/issue/twbs/bootstrap/27330"
                    onClick={[Function]}
                >
                      
                    [v3] navbar-fixed-* don't get right padding on modal open
                </a>
            </div>
            <span
                className="issue-desc"
            >
                #
                27330
                 
                opened
                 
                8 days ago
                 by 
                iamandrewluca
                <i
                    className="fa fa-comments pl-1"
                />
                 
                0
            </span>
            <span
                className="pl-2"
            >
                Last updated at 
                8 days ago
            </span>
        </li>
    </ul>
    <ul
        className="pagination justify-content-center mt-5"
    >
        <li
            className="active page-item"
        >
            <a
                className="page-link"
                onClick={[Function]}
            >
                1
            </a>
        </li>
        <li
            className="page-item"
        >
            <a
                className="page-link"
                onClick={[Function]}
            >
                2
            </a>
        </li>
    </ul>
</div>,
]
`;
